proto/substrait/algebra.proto

```diff
@@ -773,6 +775,53 @@ message NestedLoopJoinRel {
   substrait.extensions.AdvancedExtension advanced_extension = 10;
 }
 
+message DuplicateEliminatedGetRel {
+  RelCommon common = 1;
+  ReferenceRel input = 2;
```

Comment:
I'm unsure about whether the ReferenceRel rules require the referenced input to be executed first (by being earlier on the path of a depth first search) if that's true and we need the DuplicateEliminatedGetRel on the left side it may be useful to make this a Rel which is either a ReferenceRel pointing to the left side of the tree or the full computation with a ReferenceRel on the other side.   @westonpace Do you think we need that change?

Reply:
I'm not sure I understand.  I don't think the ordering in a substrait plan needs to be related to execution ordering in any way.

Reply:
I understand the concern better now.  I still don't think the order matters though.  We have essentially crossed the bridge from "acyclic" (DAG) to "has a cycle" (graph).  Inside knowledge of the relation would be required to understand how to execute it correctly.  I'm not sure it matters which side the reference is on (left or right).